.github/actions/apply-learning-path-updates/index.js

```diff
@@ -0,0 +1,73 @@
+const actionUtils = require('../action-utils.js');
+
+const suggestionsSeparator = ',';
+const oldNewLinkSeparator = ' -> ';
+let modifiedFiles = [];
+
+function AppendModifiedFiles(path)
+{
+  modifiedFiles.push(path)
+  core.setOutput('modifiedFiles', modifiedFiles.join(' '))
+}
+
+function ReplaceOldWithNewText(content, oldText, newText)
+{
+  return content.replace(new RegExp(oldText, 'g'), newText);
```

Comment:
We should avoid creating a regex expression from an arbitrary variable here. `context.replaceAll(oldText, newText)` should work here instead